vault/identity_store_entities.go

```diff
@@ -809,6 +810,22 @@ func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntit
 		// the entity we are merging into is composed of.
 		toEntity.MergedEntityIDs = append(toEntity.MergedEntityIDs, fromEntity.ID)
 
+		// Remove entity ID as a member from all the groups it belongs, both
+		// internal and external
+		groups, err := i.MemDBGroupsByMemberEntityIDInTxn(txn, fromEntity.ID, true, false)
+		if err != nil {
+			return nil, err
+		}
+		for _, group := range groups {
+			group.MemberEntityIDs = strutil.StrListDelete(group.MemberEntityIDs, fromEntity.ID)
+			err = i.UpsertGroupInTxn(ctx, txn, group, persist && !isPerfSecondaryOrStandby)
+			if err != nil {
+				return nil, err
+			}
+
+			fromEntityGroups = append(fromEntityGroups, group)
+		}
+
 		// Delete the entity which we are merging from in MemDB using the same transaction
 		err = i.MemDBDeleteEntityByIDInTxn(txn, fromEntity.ID)
```

Comment:
Unfortunately we cannot reuse `IdentityStore#handleEntityDeleteCommon` because that also removes entity aliases on the `fromEntity`. Alternatively, I believe I could do something like this instead:

```go
fromEntity.Aliases = []*Alias{}
err = i.handleEntityDeleteCommon(ctx, txn, fromEntity, persist && !isPerfSecondaryOrStandby)
if err != nil {
	return nil, err
}
```